mathics/builtin/assignment.py

```diff
@@ -64,7 +64,7 @@ def get_symbol_list(list, error_callback):
 class _SetOperator(object):
     def assign_elementary(self, lhs, rhs, evaluation, tags=None, upset=False):
         # TODO: This function should be splitted and simplified
-
+        evaluation.cache_result = False
```

Comment:
This line of code appear many many times. What's the rule, and reason behind when to do and when not to? 

I wonder if there's a better way.

Reply:
Again, this is just an experiment to see how we can implement the Expression cache.  
```
evaluation.cache_result = False
```
means that the result of this expression should not be stored in the cache. For example, control expressions,  loops and file handling should not be stored in the cache.

Reply:
Ok - thanks for the information. I got the gist of that aspect.

Generally one thinks of a cache as just working not something where you have to make 30 hand decisions as to what goes in and what doesn't. 

That's what I mean by I wonder if there is a better way.  Like using someone _else_'s mechanism like LRUCache. (Hmm, where have I heard this idea before - use someone else's package rather than rolling your own?) 

Since this is an experiment let's mark it as a draft for now.

Disclaimer: for now I am rewriting/revising the doc system. Although I would love to ditch that, I also want the docs to get tagged and organized better and using a more standard doc system is too big a lift for me right now.

Reply:
> 
> That's what I mean by I wonder if there is a better way. Like using someone _else_'s mechanism like LRUCache. (Hmm, where have I heard this idea before - use someone else's package rather than rolling your own?)
> 

OK, I totally agree: if there is something already implemented, let's use it. The problem here is that I didn't find something like LRUCache that works with expressions.  In any case, right now I do not have much time to work on this. I just rebase this in case we want to come back to it in the future.
 


> Since this is an experiment let's mark it as a draft for now.
> 
> Disclaimer: for now I am rewriting/revising the doc system. Although I would love to ditch that, I also want the docs to get tagged and organized better and using a more standard doc system is too big a lift for me right now.

I think this is more urgent: to have a clear code and documentation will help to get more help...

Reply:
> > That's what I mean by I wonder if there is a better way. Like using someone _else_'s mechanism like LRUCache. (Hmm, where have I heard this idea before - use someone else's package rather than rolling your own?)
> 
> OK, I totally agree: if there is something already implemented, let's use it. The problem here is that I didn't find something like LRUCache that works with expressions. In any case, right now I do not have much time to work on this. I just rebase this in case we want to come back to it in the future.
> 
> > Since this is an experiment let's mark it as a draft for now.
> > Disclaimer: for now I am rewriting/revising the doc system. Although I would love to ditch that, I also want the docs to get tagged and organized better and using a more standard doc system is too big a lift for me right now.
> 
> I think this is more urgent: to have a clear code and documentation will help to get more help...

I _am_ working on it and I hope to have the code for Django in place this week. Basically I'm continuing what I started before in on the Mathics side where we are grouping builtins - eventually according to how it appears in Mathematica 5 or listed as "Guides" in the online docs. So far it has been pleasing to see. 

After the code to see this on Django is done, on the Core side we can move more stuff around and make it more closely conform to the Mathematica organization.